test/GeneratedInjectorDelegatorTest.php

```diff
@@ -24,7 +24,7 @@ public function testProvidedNamespaceIsNotAString()
             ->willReturn(true)
             ->shouldBeCalledTimes(1);
         $container->get('config')
-            ->willReturn(['dependencies' => ['auto' => ['aot' => ['namespace' => []]]]])
+            ->willReturn(['dependencies' => ['auto' => ['aot' => ['namespace' => 123]]]])
```

Comment:
Maybe I should revert this change and we should avoid to use `empty` in the `GeneratedInjectorDelegator` to check against an existing namespace.

Reply:
This was introduced on purpose since a Stringification of an array (that might come from a misconfiguration) would result in `"Array"` which would be a valid namespace name but possibly not be the user's intention. This test covers this case.

See: https://github.com/zendframework/zend-di/pull/51#discussion_r358204946

Reply:
I see the problem at this point.
We want to check if the content of `namespace` is a string.
But the problem is that in case of an empty array we already fall back to a default by validation.
I would suggest that we only set the default if `namespace` is not given and otherwise validate the content.

Reply:
maybe something like: 
````php
$namespace = $aotConfig['namespace'] ?? 'Laminas\Di\Generated';

if (! is_string($namespace)) {
    throw new InvalidServiceConfigException('Provided namespace is not a string.');
}
````

Reply:
> We want to check if the content of namespace is a string.

Actually we want to check that a PHP array - even an empty one - (from a misgonfiguation) does not become `"Array"` **and** causes an exception, since this cannot be caught by static analysis at the moment.

Personally I'd prefer to convert the AOT config array into a strongly typed object instead of accessing the array directly so that static analysis and config loading catches them instead of the factory. But that exceeds the scope of this PR.